apollo-router/src/uplink/license_enforcement.rs

```diff
@@ -501,22 +501,48 @@ impl Display for LicenseEnforcementReport {
     }
 }
 
-/// License controls availability of certain features of the Router. It must be constructed from a base64 encoded JWT
+/// These are claims extracted from the License and represent ways Apollo limits the router's usage It must be constructed from a base64 encoded JWT
 /// This API experimental and is subject to change outside of semver.
 #[derive(Debug, Clone, Default)]
 pub struct License {
     pub(crate) claims: Option<Claims>,
 }
 
+/// Transactions Per Second limits. We talk as though this will be in seconds, but the Duration
+/// here is actually given to us in milliseconds via the License's JWT's claims
+#[derive(Builder, Copy, Clone, Debug, Eq, PartialEq)]
+pub(crate) struct TpsLimit {
+    capacity: usize,
+    duration: Duration,
+}
+
+// FIXME: better defaults set with some producty insight and move then to consts
+impl Default for TpsLimit {
+    fn default() -> Self {
+        Self {
+            capacity: 100,
+            duration: Duration::from_millis(1000),
+        }
+    }
+}
+
+// TODO: better docs
+/// License controls availability of certain features of the Router.
+#[derive(Debug, Builder, Copy, Clone, Default, Eq, PartialEq)]
+pub(crate) struct LicenseLimits {
+    /// Transaction Per Second limits
+    tps: TpsLimit,
+}
+
 /// Licenses are converted into a stream of license states by the expander
 #[derive(Debug, Copy, Clone, Eq, PartialEq, Default, Display)]
 pub(crate) enum LicenseState {
     /// licensed
-    Licensed,
+    Licensed(Option<LicenseLimits>),
```

Comment:
Should we do the opposite instead?

E.g. `LicenseLimits` is always present, but `tps` should be an `Option<TpsLimit>`.

Reply:
ahh, yeah! I replied above and this makes me think that it's the right idea; I'll make TpsLimits optional in LicenseLimits and use _that_ as the default

Reply:
updated